### PR TITLE
feat(tier4_autoware_utils): add twist library

### PR DESCRIPTION
**Pull request by scepter914**
Signed-off-by: scepter914 <scepter914@gmail.com>

## Description

Add twist library functions for tier4_autoware_util.
These functions will be used for radar sensing package and radar fusion perception package.

When I fixed at DCO at [this PR](https://github.com/autowarefoundation/autoware.universe/pull/2049), I miss the operation.
So I make new this PR.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by scepter914** [comment]
@kenji-miyake 
I'm sorry that I miss the operation when I fixed at DCO at https://github.com/autowarefoundation/autoware.universe/pull/2049.
So I make new this PR.

File: common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp [hunk 1]
```diff
@@ -731,6 +735,49 @@ geometry_msgs::msg::Pose calcInterpolatedPose(
...
+
+inline double calcTwistNorm(const geometry_msgs::msg::Vector3 & v)
+{
+  return std::hypot(v.x, v.y, v.z);
+}
+
+inline double calcTwistNorm(const geometry_msgs::msg::Twist & twist)
+{
+  return calcTwistNorm(twist.linear);
+}
```
Comment (kenji-miyake) [inline common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp:761]: I think this wrapper isn't necessary. Let's use `calcNorm(twist.linear)` in the application code.
Also, for `Vector3`, the word `Twist` is unnecessary.

```suggestion
inline double calcNorm(const geometry_msgs::msg::Vector3 & v)
{
  return std::hypot(v.x, v.y, v.z);
}
```

File: common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp [hunk 2]
```diff
@@ -731,6 +735,49 @@ geometry_msgs::msg::Pose calcInterpolatedPose(
...
+  return geometry_msgs::build<geometry_msgs::msg::Vector3>().x(x).y(y).z(z);
+}
+
+inline geometry_msgs::msg::Twist createTwist(
+  const geometry_msgs::msg::Vector3 & velocity, geometry_msgs::msg::Vector3 & angular)
+{
+  geometry_msgs::msg::Twist twist;
+  twist.linear = velocity;
+  twist.angular = angular;
+  return twist;
```
Comment (kenji-miyake) [inline common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp:750]: [nits] I believe you can use `geometry_msgs::build` also here. It enables us to ensure all fields of the message are initialized.

File: common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp (hunk 1 above)
Comment (scepter914) [reply common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp]: fixed at [bfcd5be](https://github.com/autowarefoundation/autoware.universe/pull/2187/commits/bfcd5be2df96b3a97c77e39947cd0c6c84714b3a)

File: common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp (hunk 2 above)
Comment (scepter914) [reply common/tier4_autoware_utils/include/tier4_autoware_utils/geometry/geometry.hpp]: fixed at [7ab88dc](https://github.com/autowarefoundation/autoware.universe/pull/2187/commits/7ab88dc7f3f73b3ed158a8110caabb72891f9e31)

**Review by kenji-miyake — APPROVED** [review]
What a clean code!

**Pull request merged**